src/components/button/Button.tsx

```diff
@@ -46,22 +44,17 @@ const ButtonTag = styled(Tag)`
 // Setting the line-height makes sure regular sized buttons have a
 // height consistent with the input fields and across variants.
 // Using the same line height as in the blueprint docs.
-function getStyledCss(
-  buttonProps: BlueprintAnchorButtonProps | BlueprintButtonProps,
-): CSSObject {
-  return {
-    lineHeight: 1.15,
-    position: 'relative',
-    [`.${Classes.ICON}`]: !buttonProps.children && { marginRight: 0 },
```

Comment:
This line is what was creating an incompatibility when using CSS layers for the blueprintjs stylesheets.

It would take precedence over blueprintjs styles because not in a layer.